### PR TITLE
refactor: add featureFlags to SystemContext

### PR DESCRIPTION
**Pull request by mc-jones**
This PR simplifies the Unleash implementation by removing the previously used custom React `Context.Provider` in favor of `Force`'s `SystemContext`. 

## Related PRs
- https://github.com/artsy/force/pull/9498
- https://github.com/artsy/force/pull/9393

## Follow-ups
- Allow for experimentation by implementing `useVariant` and analytics tracking. [PLATFORM-3922]

[PLATFORM-3922]: https://artsyproduct.atlassian.net/browse/PLATFORM-3922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/featureFlags/FeatureFlagProvider.tsx
Comment (damassi) [inline src/lib/featureFlags/FeatureFlagProvider.tsx:6]: This file can be moved to `v2/System`

File: src/lib/featureFlags/FeatureFlagProvider.tsx
Comment (damassi) [inline src/lib/featureFlags/FeatureFlagProvider.tsx:18]: I think this logic might be slightly simplified, and we can lean on typescript here as well for additional protection. 

Would suggest something like:

```tsx
export function useFeatureFlag(flagName: string): string | null {
  const { featureFlags } = useSystemContext()
  
  if (!featureFlags) {
    console.error("[Force] Error: featureFlags is undefined in SystemContext")
    return null
  }
  
  const flagName = featureFlags[flagName]
  
  if (!flagName) {
    console.error("[Force] Error: cannot find flagName in featureFlags: " Object.keys(featureFlags))
  }
  
  return flagName
}
```

**Review by damassi — DISMISSED** [review]
Looks great @mc-jones 👍 . One suggestion / relocation request around `useFeatureFlag.tsx`, and it'd be nice to see a basic test for the hook, but otherwise LGTM 👍

File: src/v2/System/Router/buildClientApp.tsx
Comment (damassi) [inline src/v2/System/Router/buildClientApp.tsx:116]: this is minor and non-blocking, but all of our other ENV variables, even set from sharify, are `FEATURE_FLAGS`, and it would be good to remain consistent here across the board.

File: src/lib/featureFlags/FeatureFlagProvider.tsx
Comment (mc-jones) [reply src/lib/featureFlags/FeatureFlagProvider.tsx:18]: Have added the typing, but going to hold off on changing this too much, as I think it may change not insignificantly as we add in the `useVariant` [functionality](https://docs.getunleash.io/advanced/toggle_variants).

**Comment by mc-jones** [comment]
> Looks great @mc-jones 👍 . One suggestion / relocation request around `useFeatureFlag.tsx`, and it'd be nice to see a basic test for the hook, but otherwise LGTM 👍

@damassi - Totally makes sense. ✔️ .

File: src/v2/System/SystemContext.tsx
Comment (damassi) [inline src/v2/System/SystemContext.tsx:82]: @mc-jones - didn't catch this before merge, but one tiny little nbd follow up is this could use the `EnabledFeatureFlags` type that was added below and shared with `Boot`

**Pull request merged**
